src/Compilers/Core/Portable/SourceCodeKind.cs

```diff
@@ -32,6 +32,24 @@ public enum SourceCodeKind
 
     internal static partial class SourceCodeKindExtensions
     {
+        internal static SourceCodeKind MapSpecifiedToEffectiveKind(this SourceCodeKind kind)
+        {
+            switch (kind)
+            {
+                case SourceCodeKind.Regular:
+                    return SourceCodeKind.Regular;
+
+                case SourceCodeKind.Script:
+#pragma warning disable CS0618 // SourceCodeKind.Interactive is obsolete
+                case SourceCodeKind.Interactive:
+#pragma warning restore CS0618 // SourceCodeKind.Interactive is obsolete
+                    return SourceCodeKind.Script;
+
+                default:
+                    throw new NotSupportedException($"SourceCodeKind {kind} not supported");
```

Comment:
>throw new NotSupportedException($"SourceCodeKind {kind} not supported"); [](start = 20, length = 72)

I thought we decided to not throw while constructing ParseOptions. Consider using SourceCodeKind.Regular as a fallback kind instead.